public/app/plugins/datasource/cloudwatch/components/MetricStatEditor/MetricStatEditor.test.tsx

```diff
@@ -99,4 +100,87 @@ describe('MetricStatEditor', () => {
       expect(screen.getByLabelText('Match exact - optional')).not.toBeChecked();
     });
   });
+
+  describe('validating Query namespace / metricName', () => {
+    const namespaces = [
+      { value: 'n1', label: 'n1', text: 'n1' },
+      { value: 'n2', label: 'n2', text: 'n2' },
+    ];
+    const metrics = [
+      { value: 'm1', label: 'm1', text: 'm1' },
+      { value: 'm2', label: 'm2', text: 'm2' },
+    ];
+
+    it('should select namespace and metric name correctly', async () => {
+      const onChange = jest.fn();
+      const onRunQuery = jest.fn();
+
+      props.datasource.getNamespaces = jest.fn().mockResolvedValue(namespaces);
+      props.datasource.getMetrics = jest.fn().mockResolvedValue(metrics);
```

Comment:
This is modifying the `props.datasource` for other tests (you need to copy it)

Reply:
Good point, I'll add that do the describe, and could probably make better use of the `beforeEach` there too 👍